src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -247,39 +249,103 @@ def create_run_endpoint_handler(pipeline_wrapper: BasePipelineWrapper, RunReques
 
     Args:
         pipeline_wrapper: The pipeline wrapper instance
-        RunRequest: The request model
-        RunResponse: The response model
+        request_model: The request model
+        response_model: The response model
         requires_files: Whether the pipeline requires file uploads
     """
 
     @handle_pipeline_exceptions()
-    async def run_endpoint_with_files(run_req: RunRequest = Form(...)) -> RunResponse:  # type: ignore
+    async def run_endpoint_with_files(run_req: request_model = Form(...)) -> response_model:  # type: ignore
         result = await run_in_threadpool(pipeline_wrapper.run_api, **run_req.model_dump())  # type: ignore
-        return RunResponse(result=result)
+        return response_model(result=result)
 
     @handle_pipeline_exceptions()
-    async def run_endpoint_without_files(run_req: RunRequest) -> RunResponse:  # type: ignore
+    async def run_endpoint_without_files(run_req: request_model) -> response_model:  # type: ignore
         result = await run_in_threadpool(pipeline_wrapper.run_api, **run_req.model_dump())  # type: ignore
-        return RunResponse(result=result)
+        return response_model(result=result)
 
     return run_endpoint_with_files if requires_files else run_endpoint_without_files
 
 
+def add_pipeline_api_route(app: FastAPI, pipeline_name: str, pipeline_wrapper: BasePipelineWrapper) -> None:
```

Comment:
Could you please explain the changes in this file?

Reply:
As we discussed, this refactoring was needed since when we start MCP server, we don't need to add any dynamic custom API route to the underlying Starlette app. So I've decoupled deploy in two steps: 1. deploy to registry 2. add_api_route